.travis.yml

```diff
@@ -2,6 +2,13 @@
 sudo: false
 matrix:
   include:
+    - env: ACTION=checkformatting
+      language: python
+      python: '3.6'
+      install: pip install tox tox-pip-extensions
```

Comment:
I worry a little that having too many entries in the build matrix will end up slowing down the build cycle as Travis limits the number that get executed concurrently, and there is a bunch of startup overhead per build. We could resolve this by consolidating the non-test tasks into a single build job.

Reply:
Doing less stuff on Travis and Jenkins might help too. Getting to Python 3 would help with that. We may also be able to speed up `make lint` by disabling all purely code-formatting rules now that formatting is automatic. I haven't really gotten round to looking at our `.travis.yml` files but I'm sure they could be improved, and also made consistent across projects